### PR TITLE
Refactor MLOpsContext into separate catalogs

### LINKED ISSUE #104: Separate MLOpsContext into catalogues
**Is your feature request related to a problem? Please describe.**
The MLOpsContext is quickly growing. To ensure we have a structure we may want to separate it out into a structure similar to MLContext.

**Describe the solution you'd like**
The MLContext in ML.NET has clear catalogs where one can find a feature. I suggest we break the functionality into separate catalogues that live as properties on the MLOpsContext

- Experiment (create/get experiments)
- Run (create/get runs)
- Data (can be used to track the data used for training later)
- Evaluation (log metrics etc.)
- Model (upload model, download model...)
- Transform/Training? (track hyper-parameters)

### PR DESCRIPTION
**Pull request by aslotte**
## Resolves
Fixes #104 

## Description
Refactored MLOpsContext into separate catalogs.

### CODE REVIEW

**Comment by aslotte** [comment]
This is ready for a review!

File: src/MLOps.NET/MLOpsContext.cs [hunk 1]
```diff
@@ -1,107 +1,34 @@
...
+            this.Training = new TrainingCatalog(metaDataStore);
         }
 
-        ///<inheritdoc/>
-        public async Task<Guid> CreateRunAsync(Guid experimentId)
-        {
-            return await metaDataStore.CreateRunAsync(experimentId);
-        }
+        ///<inheritdoc cref="IMLOpsContext"/>
+        public LifeCycleCatalog LifeCycleCatalog { get; set; }
```
Comment (shanranm) [inline src/MLOps.NET/MLOpsContext.cs:22]: Do we want to expose setters also?

File: src/MLOps.NET/Catalogs/ModelCatalog.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        /// ctor
+        /// </summary>
+        /// <param name="modelRepository"></param>
+        public ModelCatalog(IModelRepository modelRepository)
+        {
+            this.modelRepository = modelRepository;
+        }
+
+        ///<inheritdoc/>
+        public async Task UploadModelAsync(Guid runId, string filePath)
```
Comment (shanranm) [inline src/MLOps.NET/Catalogs/ModelCatalog.cs:24]: Just `UploadAsync` will do? `mlContext.Model.UploadAsync`

File: src/MLOps.NET/MLOpsContext.cs (hunk 1 above)
Comment (aslotte) [reply src/MLOps.NET/MLOpsContext.cs]: Great catch @shanranm, I did not intend to do that. Pushed a change.

File: src/MLOps.NET/Catalogs/ModelCatalog.cs (hunk 2 above)
Comment (aslotte) [reply src/MLOps.NET/Catalogs/ModelCatalog.cs]: I like the suggestion, makes a lot of sense now when it's on the specific `ModelCatalog`. Pushed a change for it.

**Comment by AnoojNair** [comment]
@aslotte. Looks good .Just wondering about the name "LifeCycleCatalog". is it too long? How about just Experiments?

**Comment by aslotte** [comment]
Great point @AnoojNair. I meant to just call it `LifeCycle`. I started off with the name `ExperimentCatalog` but since it also contains operations on `Runs` I tried to think of a better name. Happy to think about a better name going forward. Let me know if you have any suggestions! (naming is hard)

**Pull request merged**